lib/appoptics_apm/inst/mongo.rb

```diff
@@ -168,6 +168,7 @@ def appoptics_collect(m, args)
             end
 
             AppOpticsAPM::API.trace(:mongo, report_kvs) do
+              report_kvs[:Backtrace] = AppOpticsAPM::API.backtrace if AppOpticsAPM::Config[:mongo][:collect_backtraces]
```

Comment:
here these are done before invoking the `without_appoptics` method unlike mongodb. but why not put it before the entire block?

Reply:
The backtrace needs to sent once. 
The `AppOpticsAPM::API.trace` logs an entry event and then yields. When the block finishes `AppOpticsAPM::API.trace` sends the exit event that then may include the backtrace in the kvs.
Therefore the backtraces need to be added somewhere inside this do-block, ideally before anything is called that could cause an exception.

I'm moving the line in mongo.rb.
mongo2 is still sending backtraces twice, will correct, thanks for mentioning mongo